library/blob.c

```diff
@@ -1986,23 +1986,20 @@ static int eblob_write_prepare_disk_ll(struct eblob_backend *b, struct eblob_key
  * eblob_write_prepare_disk() - allocates space for new record
  * It allocates new bases, commits headers and manages overwrites/appends.
  */
-static int eblob_write_prepare_disk(struct eblob_backend *b, struct eblob_key *key,
+static int eblob_write_prepare_disk_nolock(struct eblob_backend *b, struct eblob_key *key,
 		struct eblob_write_control *wc, uint64_t prepare_disk_size,
 		enum eblob_copy_flavour copy, uint64_t copy_offset, struct eblob_ram_control *old,
 		size_t defrag_generation)
 {
 	FORMATTED(HANDY_TIMER_SCOPE, ("eblob.%u.disk.write.prepare.disk", b->cfg.stat_id));
-
 	ssize_t err = 0;
 	uint64_t size;
 	struct eblob_ram_control upd_old;
 
 	eblob_log(b->cfg.log, EBLOB_LOG_NOTICE,
-			"blob: %s: eblob_write_prepare_disk: start: "
-			"size: %" PRIu64 ", offset: %" PRIu64 ", prepare: %" PRIu64 "\n",
-			eblob_dump_id(key->id), wc->size, wc->offset, prepare_disk_size);
-
-	pthread_mutex_lock(&b->lock);
+		  "blob: %s: eblob_write_prepare_disk: start: "
+		  "size: %" PRIu64 ", offset: %" PRIu64 ", prepare: %" PRIu64 "\n",
+		eblob_dump_id(key->id), wc->size, wc->offset, prepare_disk_size);
```

Comment:
There is no your changes. Please return it to original.

Reply:
fixed